sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetFilters.scala

```diff
@@ -82,6 +120,30 @@ private[parquet] class ParquetFilters(pushDownDate: Boolean, pushDownStartWith:
       (n: String, v: Any) => FilterApi.eq(
         intColumn(n),
         Option(v).map(date => dateToDays(date.asInstanceOf[Date]).asInstanceOf[Integer]).orNull)
+
+    case ParquetSchemaType(DECIMAL, INT32, decimal) if pushDownDecimal =>
```

Comment:
Since this uses the file schema, I think it should validate that the file uses the same scale as the value passed in. That's a cheap sanity check to ensure correctness.

Reply:
Seems invalidate value already filtered by: https://github.com/apache/spark/blob/e76b0124fbe463def00b1dffcfd8fd47e04772fe/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceStrategy.scala#L439

Reply:
That doesn't validate the value against the decimal scale from the file, which is what I'm suggesting. The decimal scale must match exactly and this is a good place to check because this has the file information. If the scale doesn't match, then the schema used to read this file is incorrect, which would cause data corruption.

In my opinion, it is better to add a check if it is cheap instead of debating whether or not some other part of the code covers the case. If this were happening per record then I would opt for a different strategy, but because this is at the file level it is a good idea to add it here.

Reply:
I see. I will do it.

Reply:
Add check method to `canMakeFilterOn` and add a test case:
```scala
    val decimal = new JBigDecimal(10).setScale(scale)
    assert(decimal.scale() === scale)
    assertResult(Some(lt(intColumn("cdecimal1"), 1000: Integer))) {
      parquetFilters.createFilter(parquetSchema, sources.LessThan("cdecimal1", decimal))
    }

    val decimal1 = new JBigDecimal(10).setScale(scale + 1)
    assert(decimal1.scale() === scale + 1)

    assertResult(None) {
      parquetFilters.createFilter(parquetSchema, sources.LessThan("cdecimal1", decimal1))
    }
```